src/native/libs/System.Native/pal_crossprocessmutex_unsupported.c

```diff
@@ -0,0 +1,91 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_config.h"
+#include "pal_crossprocessmutex.h"
+#include "pal_errno.h"
+
+#include <stdio.h>
+#include <limits.h>
+#include <sched.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#ifdef DEBUG
+#define DEBUGNOTRETURN __attribute__((noreturn))
```

Comment:
Fighting with the compiler about the noreturns is lost battle with the high warning level used by System.Native.

I would rather delete `assert(false)` that the compiler is unhappy about.

Reply:
One of the earlier PRs dealing with the same issue: https://github.com/dotnet/runtime/pull/117417/